app/Helpers/Bbcode.php

```diff
@@ -7,359 +7,310 @@
-            'content' => '$1',
+            'openBbcode'  => '/^\[spoiler\]/i',
+            'closeBbcode' => '[/spoiler]',
+            'openHtml'    => '<details class="label label-primary"><summary>Spoiler</summary><pre><code><div style="text-align:left;">',
+            'closeHtml'   => '</div></code></pre></details>',
         ],
-
         'named-spoiler' => [
-            'pattern' => '/\[spoiler\=(.*?)\](.*?)\[\/spoiler\]/s',
-            'replace' => '<details class="label label-primary"><summary>$1</summary><pre><code><div style="text-align:left;">$2</div></code></pre></details>',
-            'content' => '$1',
+            'openBbcode'  => '/^\[spoiler=(.*?)\]/i',
+            'closeBbcode' => '[/spoiler]',
+            'openHtml'    => '<details class="label label-primary"><summary>$1</summary><pre><code><div style="text-align:left;">',
+            'closeHtml'   => '</div></code></pre></details>',
         ],
     ];
 
-    protected array $enabledParsers = [];
-
-    public function __construct()
-    {
-        $this->enabledParsers = $this->parsers;
-    }
-
     /**
      * Parses the BBCode string.
      */
-    public function parse($source, bool $caseInsensitive = false): string
+    public function parse($source): string
     {
-        foreach ($this->enabledParsers as $name => $parser) {
-            $pattern = ($caseInsensitive) ? $parser['pattern'].'i' : $parser['pattern'];
-
-            $source = $this->searchAndReplace($pattern, $parser['replace'], $source);
+        // Replace all void elements since they don't have closing tags
+        $source = \str_replace('[*]', '<li>', $source);
+        $source = \preg_replace_callback(
+            '/\[url\](.*?)\[\/url\]/i',
+            fn ($matches) => '<a href="'.\htmlspecialchars($matches[1]).'">'.\htmlspecialchars($matches[1]).'</a>',
+            $source
+        );
+        $source = \preg_replace_callback(
+            '/\[img\](.*?)\[\/img\]/i',
+            fn ($matches) => '<img src="'.\htmlspecialchars($matches[1]).'" loading="lazy" class="img-responsive" style="display: inline !important;">',
+            $source
+        );
+        $source = \preg_replace_callback(
+            '/\[img width=(\d+)\](.*?)\[\/img\]/i',
+            fn ($matches) => '<img src="'.\htmlspecialchars($matches[2]).'" loading="lazy" width="'.$matches[1].'px">',
+            $source
+        );
+        $source = \preg_replace_callback(
+            '/\[img=(\d+)\](.*?)\[\/img\]/i',
+            fn ($matches) => '<img src="'.\htmlspecialchars($matches[2]).'" loading="lazy" width="'.$matches[1].'px">',
+            $source
+        );
+
+        // Youtube elements need to be replaced like this because the content inside the two tags
+        // has to be moved into an html attribute
+        $source = \preg_replace_callback(
```

Comment:
Pretty sure `frameborder` is deprecated and obsolete. Can probably just remove them.

Reply:
Done